src/diffcalc_api/errors/ub.py

```diff
@@ -97,11 +97,15 @@ def __init__(self):
 class NoUbMatrixError(DiffcalcAPIException):
     """When there is no U/UB matrix, some commands in diffcalc-core fail."""
 
-    def __init__(self):
+    def __init__(self, message: Optional[str] = None):
```

Comment:
I wonder if these will work as dataclasses, doesn't affect much though

Reply:
Hmm, I guess they could. For this PR I'll just keep them as is.